### DISCUSSION TITLE
2D/3D plot for trajectories

### DISCUSSION THREAD

**Discussion by HugoFeng** [foxglove/community#208] [Q&A]
ref: https://facontidavide.github.io/PlotJuggler/visualization_howto/index.html#display-a-xy-curve
Can we have the plot panel display not only "1D" data, but also 2D or 3D trajectories.
For example in Plotjuggler, if one drag "pos.x, pos.y" into the plot, the plot will use them as axis, while the playback will update the "dot" showing the position at the moment.
This is very usefull and a must to have for working on Odometry for robotics. It'll be cooler if it can be extended to x,y,z while Plotjuggler only support x,y display.

**Comment by jtbandes** [comment]
You might be able to accomplish what you are describing using the custom x-axis settings described here: https://foxglove.dev/docs/studio/panels/plot

![image](https://user-images.githubusercontent.com/14237/157107737-97a60c3b-fcd5-4afe-840e-d2921346ac4e.png)

**Comment by HugoFeng** [comment]
> You might be able to accomplish what you are describing using the custom x-axis settings described here: https://foxglove.dev/docs/studio/panels/plot
> 
> ![image](https://user-images.githubusercontent.com/14237/157107737-97a60c3b-fcd5-4afe-840e-d2921346ac4e.png)

Thank you, it works. glad to know this! Though I have 2 more questions:
1. how can I show a marker on my current position
2. It only shows the trajectory in within the "pre-fetched" time slot. Is there anyway it can pre-load the entire "accum" trajectory?
Thanks.

**Reply by xavier-guardian** [reply]
I am also hoping for a marker that shows the current time step inside the trajectory plots. That would be amazing. I am hoping to convince my team to switch from PlotJuggler to Foxglove, and this is one of the small sticking points.

**Reply by indirectlylit** [reply]
Hi, thanks for the request!

If I understand correctly, you'd like to highlight the point(s) corresponding to the current playback time, is that correct?

If so, we are tracking that internally under issue FG-9681. No promises on timeframe but it is something we are aware would help many users.

**Reply by xavier-guardian** [reply]
Excellent! That is exactly what I would like yes. Thank you for responding!

Small naive question, where can i view the other FGs? I can't find them on the github. N00b issue.

**Reply by defunctzombie** [reply]
Those are issue numbers in our internal tracker - that is only accessible to employees but we often note it on public issues as a reference to readers and other employees that read the discussion.

**Reply by xavier-guardian** [reply]
I think this is implemented now!

Now I am missing the a highlighted point on the current timestep/cursor more than ever!!

**Reply by yahsiuhsieh** [reply]
Hey @xavier-guardian, could you show me how to achieve this? I'm having difficulty reaching the same thing. Thanks!

**Reply by indirectlylit** [reply]
@yahsiuhsieh we don't yet have the ability to highlight the current point, but it is in the works!

**Reply by yahsiuhsieh** [reply]
Awesome thank you @indirectlylit ! Could you let me know the expected time for this feature?

**Reply by indirectlylit** [reply]
I can't make promises because sometimes other work takes priority over new features like this, but I hope it will be within a month or so.

**Reply by yahsiuhsieh** [reply]
Thank you so much for your responsiveness! Looking forward to the feature!
